pkg/controller/config/config_controller.go

```diff
@@ -169,6 +169,8 @@ type ReconcileConfig struct {
 // Automatically generate RBAC rules to allow the Controller to read all things (for sync)
 // update is needed for finalizers
 func (r *ReconcileConfig) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	defer r.tracker.Observe("config.gatekeeper.sh", "Config", request.Name, request.Namespace)
```

Comment:
I don't think we necessarily want to defer observation, as an observe indicates a successful ingest, which may be interrupted by transient situations such as network errors. Also, for actions like delete, we probably want to cancel expectations. It's a subtle difference, but might be important later.

Also, in the case of config, what we are doing is making sure we are tracking the resources marked as watchable by the config, not necessarily the config itself (hence the focus on the diff).

Is there a bug in the config controller that I'm not seeing? To me it looks like the current code should work as-is.

Reply:
In this case I was indifferent to the success of ingestion. It would probably make sense to check if a Requeue was requested, before observing.

> Is there a bug in the config controller that I'm not seeing? To me it looks like the current code should work as-is.

Nope code is fine, I switched from caring about some details to simplify the code.

Why should it matter to cancel expectations? From what I can tell canceling or observing will result in the same outcome.  So in my mind it made sense to only `Observe` and avoid juggling state. I did this to appeal to my bias of keeping things simple and dumb.

Reply:
I believe for "parented" expectations (e.g. a constraint template causes the system to also load expectations for constraints), the cancellation is propagated to the child expectations (I may be misremembering). Observing merely means that the resource at-hand has been synced (e.g. constraint template has been ingested, constraints may not be).